libs/blocks/email-collection/email-collection.js

```diff
@@ -0,0 +1,493 @@
+  const emailEl = el.querySelector('.subscribed-email');
+  if (emailEl) emailEl.textContent = email;
+  emailEl?.classList.toggle('hidden', !subscribed);
+}
+
+export const [showHideMessage, setMessageEls] = (() => {
+  let elsObject = {};
+  return [
+    ({ errorMsg, subscribed = false, email, hideMessage = false }) => {
+      const { foreground, success, error, form } = elsObject;
+      [form, success, error].forEach((el) => el.classList.add('hidden'));
+
+      const replace = errorMsg ? error : success;
+      if (errorMsg) window.lana?.log(errorMsg);
+
+      foreground.classList.toggle('message', !hideMessage);
+      form.classList.toggle('hidden', !hideMessage);
+      replace.classList.toggle('hidden', hideMessage);
+
+      showHideSubscribedMessage(replace, subscribed, email);
+      updateAriaLive(replace);
+
+      if (!hideMessage) return;
+      const emailEl = form.querySelector('#email');
+      if (emailEl) emailEl.value = '';
+      emailEl?.focus();
+    },
+    (children) => {
+      const foreground = children[0];
+      elsObject = {
+        foreground,
+        form: foreground.children[1],
+        success: children[1].firstElementChild,
+        error: children[2].firstElementChild,
+      };
+      return Object.values(elsObject).every((value) => value);
+    },
+  ];
+})();
+
+async function insertProgress(el, size = 'm') {
+  if (!el) return;
+
+  const { base } = miloConfig;
+  await Promise.all([
+    import('../../deps/lit-all.min.js'),
+    import(`${base}/features/spectrum-web-components/dist/theme.js`),
+    import(`${base}/features/spectrum-web-components/dist/progress-circle.js`),
+  ]);
+  let staticColor;
+  if (el.tagName === 'BUTTON') {
+    const { width, height } = window.getComputedStyle(el);
+    el.style.width = width;
+    el.style.height = height;
+    if (el.matches('.con-button.outline')) staticColor = 'white';
+  }
+  const theme = createTag(
+    'sp-theme',
+    {
+      system: 'spectrum',
+      color: 'light',
+      scale: 'medium',
+      class: 'progress-circle-container',
+    },
+  );
+  const progress = createTag(
+    'sp-progress-circle',
+    {
+      label: 'Loading content',
+      indeterminate: true,
+      'static-color': staticColor,
+      size,
+    },
+  );
+
+  theme.appendChild(progress);
+  el.replaceChildren(theme);
+}
+
+function decorateSelect(data, id, placeholder) {
+  const selectWrapper = createTag('div', { class: 'select-wrapper' });
+  const select = createTag('select', { id, name: id });
+  const optionPlaceholder = createTag('option', { value: '' }, placeholder?.trim());
+  select.appendChild(optionPlaceholder);
+
+  data.forEach(({ key, value }) => {
+    const option = createTag('option', { value: key }, value);
+    select.appendChild(option);
+  });
+
+  selectWrapper.appendChild(select);
+  return selectWrapper;
+}
+
+async function decorateInput(key, value) {
+  const profile = await getIMSProfile();
```

Comment:
Can it be that we don't have the profile here, aka not signed in? Would that lead to any issues?